runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -2298,6 +2299,15 @@ remoteCompile(
    std::string detailsStr = std::string((char*) &details, sizeof(TR::IlGeneratorMethodDetails));
    TR::CompilationInfo *compInfo = compInfoPT->getCompilationInfo();
    bool useAotCompilation = compInfoPT->getMethodBeingCompiled()->_useAotCompilation;
+   JITaaS::J9ClientStream *client;
+   if (enableJITaaSPerCompConn)
+      {
+      client = new (PERSISTENT_NEW) JITaaS::J9ClientStream(compInfo->getPersistentInfo());
+      }
+   else
+      {
+      client = compInfoPT->getClientStream();
```

Comment:
See my comment above about creating the connection here (but only if it doesn't exist already)